capnp/src/schema.rs

```diff
@@ -70,10 +70,7 @@ impl StructSchema {
         if let Some(field) = self.find_field_by_name(name)? {
             Ok(field)
         } else {
-            Err(crate::Error::failed(format!(
-                "field \"{}\" not found",
```

Comment:
It's sad that we lose the field name here.

This makes me think that we should:
1. Rename the `description` field to `extra` (meaning "extra context about error").
2. Add a `Error::extra(&mut self, message: String)` method for populating it. When `alloc` is not enabled, this will just drop `message`.

Reply:
the problem here is that `String` is not available in `no alloc` scenarios, so I need to guard the function itself with `#[cfg(feature = "alloc")]`, and also every call site; but if here is the only case where it's needed, it shouldn't be too bad